packages/emitter-framework/src/python/components/function-declaration/function-declaration.tsx

```diff
@@ -0,0 +1,86 @@
+import { useTsp } from "#core/index.js";
+import { buildParameterDescriptors } from "#python/utils/operation.js";
+import { declarationRefkeys } from "#python/utils/refkey.js";
+import * as py from "@alloy-js/python";
+import type { Model, Operation } from "@typespec/compiler";
+import { createDocElement } from "../../utils/doc.js";
+import { TypeExpression } from "../type-expression/type-expression.js";
+
+export interface FunctionDeclarationPropsWithType
+  extends Omit<py.FunctionDeclarationProps, "name"> {
+  type: Operation;
+  name?: string;
+  parametersModel?: Model;
+  /** If true, parameters replaces operation parameters instead of adding to them as keyword-only */
+  replaceParameters?: boolean;
+}
+
+export type FunctionDeclarationProps =
+  | FunctionDeclarationPropsWithType
+  | py.FunctionDeclarationProps;
+
+/**
+ * A Python function declaration. Pass the `type` prop to create the
+ * function declaration by converting from a TypeSpec Operation. Any other props
+ * provided will take precedence.
+ *
+ * Behavior notes:
```

Comment:
I use @remarks for this, at some point @remarks will render nicely in TypeSpec docs (as it does for Alloy docs)